receiver/webhookeventreceiver/config.go

```diff
@@ -20,12 +20,13 @@ var (
 
 // Config defines configuration for the Generic Webhook receiver.
 type Config struct {
-	confighttp.ServerConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
-	ReadTimeout             string                   `mapstructure:"read_timeout"`    // wait time for reading request headers in ms. Default is 500ms.
-	WriteTimeout            string                   `mapstructure:"write_timeout"`   // wait time for writing request response in ms. Default is 500ms.
-	Path                    string                   `mapstructure:"path"`            // path for data collection. Default is /events
-	HealthPath              string                   `mapstructure:"health_path"`     // path for health check api. Default is /health_check
-	RequiredHeader          RequiredHeader           `mapstructure:"required_header"` // optional setting to set a required header for all requests to have
+	confighttp.ServerConfig    `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	ReadTimeout                string                   `mapstructure:"read_timeout"`                  // wait time for reading request headers in ms. Default is 500ms.
+	WriteTimeout               string                   `mapstructure:"write_timeout"`                 // wait time for writing request response in ms. Default is 500ms.
+	Path                       string                   `mapstructure:"path"`                          // path for data collection. Default is /events
+	HealthPath                 string                   `mapstructure:"health_path"`                   // path for health check api. Default is /health_check
+	RequiredHeader             RequiredHeader           `mapstructure:"required_header"`               // optional setting to set a required header for all requests to have
+	ConvertHeadersToAttributes bool                     `mapstructure:"convert_headers_to_attributes"` // optional to convert all headers to attributes
```

Comment:
Can we make this a regex instead? If empty, that setting doesn't apply. Otherwise, any header key mapping to the regular expression can be included. `.*` satisfies your use case.

Reply:
That does sound cleaner, would also help us filter out some unrelated headers that are just getting scooped up